src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -380,37 +404,244 @@ public static Translation2d[] getModuleTranslations() {
+
+    ChassisSpeeds speedsRobotRelative =
+        ChassisSpeeds.fromFieldRelativeSpeeds(speedsFieldRelative, getRotation());
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                speedsRobotRelative.vxMetersPerSecond * time,
+                speedsRobotRelative.vyMetersPerSecond * time,
+                Rotation2d.fromRadians(speedsRobotRelative.omegaRadiansPerSecond * time)));
+  }
+
+  /**
+   * Gets the pose at some time in the future, assuming constant velocity and uses robot's current
+   * speeed
+   *
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getLinearFuturePose(double time) {
+    return getLinearFuturePose(time, getVelocity());
+  }
+
+  /**
+   * Gets the pose at some time in the future, assuming constant velocity Uses fixed lookahead time
+   * specified in AutoAim.java
+   *
+   * @return The future pose
+   */
+  @AutoLogOutput(key = "AutoAim/FuturePose")
+  public Pose2d getLinearFuturePose() {
+    return getLinearFuturePose(AutoAim.LOOKAHEAD_TIME);
+  }
+
+  /**
+   * Faces the robot towards a translation on the field Keeps the robot in a linear drive motion for
+   * time seconds while rotating
+   *
+   * @param xMetersPerSecond Requested X velocity
+   * @param yMetersPerSecond Requested Y velocity
+   * @param time Time in the future to point from
+   * @return A command refrence that rotates the robot to a computed rotation
+   */
+  public Command teleopPointTowardsTranslationCmd(
+      DoubleSupplier xMetersPerSecond, DoubleSupplier yMetersPerSecond, double time) {
+    ProfiledPIDController headingController =
+        // assume we can accelerate to max in 2/3 of a second
+        new ProfiledPIDController(
+            40.0, 0.0, 0.0, new Constraints(MAX_ANGULAR_SPEED, MAX_ANGULAR_SPEED / 0.666666));
+    headingController.enableContinuousInput(-Math.PI, Math.PI);
+
+    return Commands.sequence(
+        Commands.runOnce(
+            () -> {
+              
+              this.inputSpeeds =
+                  new ChassisSpeeds(
+                      xMetersPerSecond.getAsDouble(),
+                      yMetersPerSecond.getAsDouble(),
+                      getVelocity().omegaRadiansPerSecond);
+              this.virtualTarget =
+                  getVirtualTarget(
+                      ChassisSpeeds.fromFieldRelativeSpeeds(this.inputSpeeds, getRotation()));
+              this.rotationToTranslation =
+                  getRotationToTranslation(this.virtualTarget, this.inputSpeeds)
+                      .minus(Rotation2d.fromDegrees(180));
+            },
+            this),
+        this.runVelocityFieldRelative(
+                () -> {
+                  double feedbackOutput =
+                      headingController.calculate(
+                          getPose().getRotation().getRadians(),
+                          this.rotationToTranslation.getRadians());
```

Comment:
It is definitely a bit confusing, but the `enableContinuousInput` on the lines above treats it more as a circle and will just wrap around from pi to -pi instead of a full turn.